### PR TITLE
[Deploy] Update mainnet config & add deployment artifact

### PR DESCRIPTION
**Pull request by ducthotran2010**
### Description
Mainly prepare configuration for mainnet

To deploy on Ronin mainnet:
```sh
yarn hardhat deploy --network ronin-mainnet --tags RoninTrustedOrganizationProxy,MaintenanceProxy,StakingVestingProxy,SlashIndicatorProxy,StakingProxy,RoninValidatorSetProxy,BridgeTrackingProxy,RoninGatewayPauseEnforcer
```

To deploy on Ethereum:
```sh
yarn hardhat deploy --network ethereum --tags MainchainGovernanceAdmin,MainchainRoninTrustedOrganizationProxy,MainchainGatewayPauseEnforcer
```

### Checklist
- [x] I have clearly commented on all the main functions following the [NatSpec Format](https://docs.soliditylang.org/en/v0.8.0/natspec-format.html)
- [x] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works

### CODE REVIEW

File: src/configs/config.ts [hunk 1]
```diff
@@ -58,6 +58,11 @@ export const generalRoninConf: GeneralConfig = {
     startedAtBlock: 11710199,
     bridgeContract: '0xCee681C9108c42C710c6A8A949307D5F13C9F3ca',
   },
+  [Network.Mainnet]: {
+    roninChainId: 2020,
+    startedAtBlock: 0, // TODO: update this variable
```
Comment (ducthotran2010) [inline src/configs/config.ts:63]: @minh-bq

File: src/configs/addresses.ts [hunk 2]
```diff
@@ -107,11 +107,45 @@ const trustedOrgGoerliForDevnetConfig: TrustedOrganizationStruct[] = [
...
+  governor: addr,
+  bridgeVoter: addr,
+  weight: 100,
+  addedBlock: 0,
+}));
+
+const mainnetTrustedOrgConfig: TrustedOrganizationStruct[] = [].map((addr) => ({
+  consensusAddr: addr,
+  governor: addr,
+  bridgeVoter: addr,
```
Comment (minh-bq) [inline src/configs/addresses.ts:133]: These 3 addresses should be different

File: src/configs/addresses.ts [hunk 3]
```diff
@@ -107,11 +107,45 @@ const trustedOrgGoerliForDevnetConfig: TrustedOrganizationStruct[] = [
...
+  '0xb212F24D850a0Ed90F2889dee31870E7FF3fED6c',
+].map((addr) => ({
+  consensusAddr: addr,
+  governor: addr,
+  bridgeVoter: addr,
+  weight: 100,
+  addedBlock: 0,
+}));
+
+const mainnetTrustedOrgConfig: TrustedOrganizationStruct[] = [].map((addr) => ({
```
Comment (ducthotran2010) [inline src/configs/addresses.ts:130]: @minh-bq

File: src/configs/config.ts [hunk 4]
```diff
@@ -109,7 +120,9 @@ export const stakingConfig: StakingConfig = {
     ...defaultStakingConf,
     minValidatorStakingAmount: BigNumber.from(10).pow(18).mul(BigNumber.from(10).pow(5).mul(5)), // 500.000 RON
   },
-  [Network.Mainnet]: undefined,
+  [Network.Mainnet]: {
+    ...defaultStakingConf, // TODO: double check with PO
```
Comment (minh-bq) [inline src/configs/config.ts:124]:
```suggestion
    ...defaultStakingConf, // TODO: double check with PO
    minValidatorStakingAmount: BigNumber.from(10).pow(18).mul(250_000), // 250.000 RON
```

File: src/configs/config.ts [hunk 5]
```diff
@@ -176,7 +191,9 @@ export const slashIndicatorConf: SlashIndicatorConfig = {
       jailDurationForUnavailabilityTier2Threshold: 2 * 28800, // jails for 2 days
     },
   },
-  [Network.Mainnet]: undefined,
+  [Network.Mainnet]: {
+    ...defaultSlashIndicatorConf, // TODO: double check with PO
```
Comment (minh-bq) [inline src/configs/config.ts:195]:
```suggestion
    ...defaultSlashIndicatorConf, // TODO: double check with PO
    creditScore: {
      gainCreditScore: 50,
      maxCreditScore: 600,
      bailOutCostMultiplier: 2,
      cutOffPercentageAfterBailout: 50_00, // 50%
    },
```

File: src/configs/config.ts (hunk 1 above)
Comment (minh-bq) [reply src/configs/config.ts:63]: Ronin code currently sets this to 23157000
https://explorer.roninchain.com/block/23157000

File: src/configs/config.ts [hunk 6]
```diff
@@ -91,7 +100,9 @@ export const maintenanceConf: MaintenanceConfig = {
     ...defaultMaintenanceConf,
     minMaintenanceDurationInBlock: 200,
   },
-  [Network.Mainnet]: undefined,
+  [Network.Mainnet]: {
+    ...defaultMaintenanceConf, // TODO: double check with PO
```
Comment (minh-bq) [inline src/configs/config.ts:104]:
```suggestion
    ...defaultMaintenanceConf, // TODO: double check with PO
    minOffsetToStartSchedule: 1000,
```

File: src/configs/config.ts [hunk 7]
```diff
@@ -271,7 +253,16 @@ export const roninTrustedOrganizationConf: RoninTrustedOrganizationConfig = {
     numerator: 9,
     denominator: 11,
   },
-  [Network.Ethereum]: undefined,
+  [Network.Mainnet]: {
+    trustedOrganizations: trustedOrgSet[Network.Mainnet],
+    numerator: 9,
+    denominator: 11,
```
Comment (minh-bq) [inline src/configs/config.ts:259]:
```suggestion
    denominator: 12,
```

File: src/configs/config.ts [hunk 8]
```diff
@@ -271,7 +253,16 @@ export const roninTrustedOrganizationConf: RoninTrustedOrganizationConfig = {
...
+  },
+  [Network.Ethereum]: {
+    trustedOrganizations: trustedOrgSet[Network.Ethereum],
+    numerator: 9,
+    denominator: 11,
```
Comment (minh-bq) [inline src/configs/config.ts:264]:
```suggestion
    denominator: 12,
```

**Review by minh-bq — CHANGES_REQUESTED** [review]

**Comment by nxqbao** [comment]
Everything looks good after suggestions.

File: src/configs/config.ts (hunk 7 above)
Comment (ducthotran2010) [reply src/configs/config.ts:259]:
```suggestion
    denominator: 12, // TODO: Double check with PO
```

File: src/configs/config.ts (hunk 8 above)
Comment (ducthotran2010) [reply src/configs/config.ts:264]:
```suggestion
    denominator: 12, // TODO: Double check with PO
```

File: src/configs/config.ts [hunk 9]
```diff
@@ -118,15 +131,19 @@ const defaultStakingVestingConf: StakingVestingArguments = {
...
   [Network.Devnet]: defaultStakingVestingConf,
   [Network.Testnet]: {
     ...defaultStakingVestingConf,
     topupAmount: BigNumber.from(10).pow(18).mul(BigNumber.from(10).pow(5)), // 100.000 RON
   },
-  [Network.Mainnet]: undefined,
+  [Network.Mainnet]: {
+    ...defaultStakingVestingConf,
+    blockProducerBonusPerBlock: BigNumber.from('2853881278540000000'), // 2.853_881_278_540_000_000 RON per block
+    bridgeOperatorBonusPerBlock: BigNumber.from('95129375950000000'), // 0.095_129_375_950_000_000 RON per block
```
Comment (ducthotran2010) [inline src/configs/config.ts:144]:
```suggestion
    blockProducerBonusPerBlock: BigNumber.from('2853881278540000000'), // 2.85388127854 RON per block
    bridgeOperatorBonusPerBlock: BigNumber.from('95129375950000000'), // 0.09512937595 RON per block
```

Comment (ducthotran2010) [reply src/configs/config.ts:144]: More detail at: https://sky-mavis.slack.com/archives/C03SS9CPV28/p1679630500295939?thread_ts=1679630471.807099&cid=C03SS9CPV28

File: src/configs/config.ts [hunk 10]
```diff
@@ -200,7 +243,15 @@ export const roninValidatorSetConf: RoninValidatorSetConfig = {
     numberOfBlocksInEpoch: 200,
     minEffectiveDaysOnwards: 7,
   },
-  [Network.Mainnet]: undefined,
+  [Network.Mainnet]: {
+    maxValidatorNumber: 22,
+    maxPrioritizedValidatorNumber: 12,
+    maxValidatorCandidate: 100,
```
Comment (ducthotran2010) [inline src/configs/config.ts:249]:
```suggestion
    maxValidatorCandidate: 50,
```

File: src/configs/gateway.ts [hunk 11]
```diff
@@ -192,6 +193,12 @@ export const gatewayPauseEnforcerConf: GatewayPauseEnforcerConfig = {
     enforcerAdmin: '0x968d0cd7343f711216817e617d3f92a23dc91c07',
     sentries: ['0x968D0Cd7343f711216817E617d3f92a23dC91c07'],
   },
-  [Network.Mainnet]: undefined,
-  [Network.Ethereum]: undefined,
+  [Network.Mainnet]: {
+    enforcerAdmin: '0x8417AC6838be147Ab0e201496B2E5eDf90A48cC5', // https://explorer.roninchain.com/address/ronin:8417AC6838be147Ab0e201496B2E5eDf90A48cC5
+    sentries: [],
```
Comment (ducthotran2010) [inline src/configs/gateway.ts:198]:
```suggestion
    sentries: ['0x8B35C5E273525a4Ca61025812f29C17727948f57'],
```

File: src/configs/gateway.ts [hunk 12]
```diff
@@ -192,6 +193,12 @@ export const gatewayPauseEnforcerConf: GatewayPauseEnforcerConfig = {
...
+  },
+  [Network.Ethereum]: {
+    enforcerAdmin: '0x2DA02aC5f19Ae362a4121718d990e655eB628D96', // https://etherscan.io/address/0x2DA02aC5f19Ae362a4121718d990e655eB628D96
+    sentries: [],
```
Comment (ducthotran2010) [inline src/configs/gateway.ts:202]:
```suggestion
    sentries: ['0x8B35C5E273525a4Ca61025812f29C17727948f57'],
```

**Review by minh-bq — APPROVED** [review]

**Review by nxqbao — APPROVED** [review]

**Comment by ducthotran2010** [comment]
# Mainnet Deployment Log
```
deploying "BridgeTrackingLogic" (tx: 0x68957a05f139a5f180c70b408a204e16580cf8c908c7372536cfbcc44c821f12)...: deployed at 0xFCA143FE2751A992e8Cff2eB5D64Eda809049295 with 1106178 gas
deploying "MaintenanceLogic" (tx: 0xf7275892638322ca78779c3b8c9718312207ba3d1c2368a99513333db0630a73)...: deployed at 0x44dCC40067233d3E45b93951184fB389c068466d with 1408139 gas
deploying "StakingVestingLogic" (tx: 0x70f246f2c0919cc0d392fc83cfa3e19300eb13fd51693e31ce057a36a2ca3bbe)...: deployed at 0xf07121671F929d89eA6CaEE8e3D26BD7eD63CFeb with 546074 gas
deploying "SlashIndicatorLogic" (tx: 0x44054c483a6f6ebbeafc81f2aee16bb6bc2498aae7662031d0bcb3bc6f5e62b7)...: deployed at 0x07e099e134997AF88D13f1C63F7CD34AB786c4E7 with 2594637 gas
deploying "StakingLogic" (tx: 0xec6aa9d9938342e29fe2640d7095a83fdc90822306bd5e85e1cc63fe5ee4f331)...: deployed at 0xe84C84C1e6c488a7947d21027CDd6674821348A1 with 3289965 gas
deploying "RoninValidatorSetLogic" (tx: 0x47f8413470f0f37599d3564f9d77984021807ff3da8966fee0650f5a80744542)...: deployed at 0x42Ee6A9aeC14Ca09f9d4F3eC44cd88f76b9b11F1 with 5376810 gas
deploying "RoninTrustedOrganizationLogic" (tx: 0xdf5e1aac15251234404eba4f7200f2569c27679c37baabbefd4b533ff6cf9241)...: deployed at 0x877A15cD0bf1667BaaB50b4C85FD79125c47b7B3 with 2119207 gas
deploying "RoninGatewayV2Logic" (tx: 0x47987321f7b4b26784e0d6b7d63abf6f37274c7b6c2538789b0063738a218e34)...: deployed at 0xe44b763dEb31b412c6c2b2Bd0FeDB7D12aCC9131 with 5161997 gas
deploying "RoninGovernanceAdmin" (tx: 0x26e7c72eaf10195d8e22ce6d8385b2cebb81c3eaf6fad462b5160f25af9d8e76)...: deployed at 0x946397deDFd2f79b75a72B322944a21C3240c9c3 with 5159101 gas
deploying "RoninTrustedOrganizationProxy" (tx: 0xca0d0ad93fd41340810f45fa811a9f7910473524210a3088cd56f8e184d0113f)...: deployed at 0x98D0230884448B3E2f09a177433D60fb1E19C090 with 967588 gas
deploying "MaintenanceProxy" (tx: 0x12c8daccc6fa4db84244f59133b3e623e537a4fceb851fa8bca4bd894ab5e106)...: deployed at 0x6F45C1f8d84849D497C6C0Ac4c3842DC82f49894 with 791668 gas
deploying "StakingVestingProxy" (tx: 0x17a606753d3f3940983c0ab318feef027d16905fad782d82a68a93e97529584e)...: deployed at 0xC768423A2AE2B5024cB58F3d6449A8f5DB6D8816 with 710098 gas
deploying "SlashIndicatorProxy" (tx: 0x7ac3bf6ae9c4be26f911eafa1388ea554d14c4c03b8bbe5a247c35753b384238)...: deployed at 0xEBFFF2b32fA0dF9C5C8C5d5AAa7e8b51d5207bA3 with 1111145 gas
deploying "StakingProxy" (tx: 0x571631ac7a3f269077e5bbdae19836dce4699a852ec5e265c9edcff0ee5e29ca)...: deployed at 0x545edb750eB8769C868429BE9586F5857A768758 with 772778 gas
deploying "RoninValidatorSetProxy" (tx: 0xc79d9df6f0ab141ddb854888f8c22727386e0ce0805ecfdb8ca56909fa0d858b)...: deployed at 0x617c5d73662282EA7FfD231E020eCa6D2B0D552f with 949399 gas
deploying "BridgeTrackingProxy" (tx: 0xd41c1fa59c8bbe36ee30d1446f4bad1d3fd771349b7ca1821b60b182b70fcda8)...: deployed at 0x3Fb325b251eE80945d3fc8c7692f5aFFCA1B8bC2 with 710064 gas
deploying "RoninGatewayPauseEnforcer" (tx: 0x317573ac95dd9c4f6c88a1fc72fddce0d27f82670b3ca625d7f35cbd58c1b745)...: deployed at 0x91164adC15251B90EE713a9F62a9B2A09C593c22 with 1133980 gas
```

**Comment by ducthotran2010** [comment]
# Ethereum Deployment Log
```
deploying "MainchainGatewayPauseEnforcer" (tx: 0x41e32b8a2e8f8668ffbcd5032847d804bc90736c0cf892f2db727c49baba4109)...: deployed at 0xe514d9DEB7966c8BE0ca922de8a064264eA6bcd4 with 1144480 gas
deploying "RoninTrustedOrganizationLogic" (tx: 0x399ee8391a34228f0a0fe88522bfe32c57de198039733cf0acc0b1af5f9b459a)...: deployed at 0x9Be8BB3C6ced4C0C51b1C943Dee26a593b1E6794 with 2119207 gas
deploying "MainchainGovernanceAdmin" (tx: 0x1e9c63cefacab81103635a8f2a6be8550330e3e67168411fe1db1e125d2359f1)...: deployed at 0xB255D6A720BB7c39fee173cE22113397119cB930 with 4569736 gas
deploying "MainchainRoninTrustedOrganizationProxy" (tx: 0x0eabf1611048baa62a0ed7b3dab06120239a760629aa9c6adfbe41beee980c89)...: deployed at 0x7D0556D55ca1a92708681e2e231733EBd922597D with 986288 gas
```

**Comment by ducthotran2010** [comment]
This pull request is missing the deployment of `MainchainGatewayV2Logic`

**Comment by ducthotran2010** [comment]
# Ethereum Deployment Log
```
deploying "MainchainGatewayV2Logic" (tx: 0xa5db6e43cbfa2db0c465d53a76090a5b9fc11355725755298a428a70c9ca92e3)...: deployed at 0xa67BA5315AF4961Eb937158032AF9300C657dAcD with 4504647 gas
```

**Pull request merged**